apps/oxlint/src/command/lint.rs

```diff
@@ -62,14 +66,49 @@ pub struct BasicOptions {
     /// Oxlint configuration file (experimental)
     ///  * only `.json` extension is supported
     ///  * tries to be compatible with the ESLint v8's format
+    ///
+    /// If not provided, Oxlint will look for the following files in the current
+    /// directory (in order):
+    ///
+    /// - `.oxlintrc.json`
+    /// - `oxlintrc.json`
+    /// - `.oxlint.json`
+    /// - `oxlint.json
     #[bpaf(long, short, argument("./oxlintrc.json"))]
-    pub config: Option<PathBuf>,
+    config: Option<PathBuf>,
 
     /// TypeScript `tsconfig.json` path for reading path alias and project references for import plugin
     #[bpaf(argument("./tsconfig.json"), hide_usage)]
     pub tsconfig: Option<PathBuf>,
 }
 
+impl BasicOptions {
+    const DEFAULT_OXLINTRC_NAMES: [&'static str; 4] =
+        [".oxlintrc.json", ".oxlint.json", "oxlintrc.json", "oxlint.json"];
+    pub fn config(&self) -> Option<Cow<'_, Path>> {
+        if let Some(config) = &self.config {
+            return Some(config.into());
+        }
+
+        let pwd = std::env::current_dir().ok()?;
+        for entry in std::fs::read_dir(&pwd).ok()? {
```

Comment:
I'm not sure about the performance overhead of this :-/

Reply:
We can change this to concat our paths and only check the fs stat of those few paths.

I'm more interested in this comment of yours:

> This behavior differs from eslint, will this give a surprise? Is implicit better than explicit here?

I feel it's useful for LSP, How does it differ from eslint? Doesn't eslint use the cwd to find the config if none is provided? In your opinion doing this for the CLI is a bad diversion?

Can't we make this work with something like this: https://eslint.org/docs/latest/use/command-line-interface#--no-eslintrc

I can think of situations where people would want to omit the config flag to run the linter with the default configuration, But most of the time that's not the case. I'm not sure about the implications of this change so please share your thoughts on it.